packages/clay-modal/src/Body.tsx

```diff
@@ -28,16 +28,43 @@ const ClayModalBody: React.FunctionComponent<IBodyProps> = ({
 	iFrameProps = {},
 	scrollable,
 	url,
-}: IBodyProps) => (
-	<div
-		className={classNames('modal-body', {
-			'inline-scroller': scrollable,
-			'modal-body-iframe': url,
-		})}
-		tabIndex={scrollable ? 0 : undefined}
-	>
-		{url ? <iframe {...iFrameProps} src={url} title={url} /> : children}
-	</div>
-);
+}: IBodyProps) => {
+	const elementRef = useRef<HTMLDivElement | null>(null);
+
+	useEffect(() => {
+		const onKeyDown = (event: KeyboardEvent) => {
+			if (
+				elementRef.current &&
+				(event.key === 'ArrowUp' || event.key === 'ArrowDown') &&
+				!elementRef.current.contains(document.activeElement)
+			) {
+				if (event.defaultPrevented) {
+					return;
+				}
+
+				elementRef.current.focus();
+			}
+		};
+
+		document.addEventListener('keydown', onKeyDown);
+
+		return () => {
+			document.removeEventListener('keydown', onKeyDown);
+		};
+	}, [elementRef]);
+
+	return (
+		<div
+			className={classNames('modal-body', {
+				'inline-scroller c-inner': scrollable,
```

Comment:
The markup should go:
```
<div class="modal-body inline-scroller" tabindex="0">
  <div class="c-inner" tabindex="-1">
  </div>
</div>
```
`c-inner` is an overlay that highjacks the click so focus outline won't show on `modal-body`. This applies to `popover` and `dropdown` also. I'll need to add styles to Clay CSS since we are going this route.

Reply:
> We still want to show the focus outline when someone tabs into modal-body. With tabindex="-1" a keyboard user can't tab into modal-body if they tab out of it.

I'd say that we do **NOT** want to show the focus outline. We want to allow tabbing inside or interacting somehow, although as I already expressed in https://github.com/liferay/clay/issues/3583, tabbing into a box sounds wrong (could be the only way, though).

Reply:
One quick hack that comes to mind would be to add an invisible tabbable element inside inside the scroller. Something like:

```
<div tabindex="0"></div>
```

This would avoid the focus ring while making the content scrollable. 

That being said, I have no idea how users handle focus. I don't see an outer focus ring in pages, so it seems natural not to have it, but at the same time, how do users know what are they scrolling? 

I'm honestly just at a loss here 🤷

Reply:
@pat270 and @jbalsas yeah, in the case of Modal I tried to simulate the same behavior that happens on a page that has a scroll, so we don't need to have a focus ring on the `inline-scroll`, the user can scroll the content by pressing the Up/Down keyboard, this is a little bit different from the page 😬, but it can help.

See this example in the storybook https://deploy-preview-3582--next-storybook-clayui.netlify.app/?path=/story/components-claymodal--default with the scrollable knob enabled.

Reply:
@matuzalemsteles I checked out the demo. I see how it works now. We don't need `c-inner`,  just use `tabindex="-1"`. I added a rule where outlines aren't shown when `tabindex` is negative.

Reply:
@pat270 I had to add `c-inner` exactly why is it appearing outline, maybe there was a regression?

Reply:
Strange, it works fine on the storybook you linked. I see the rule there:
```
[tabindex^='-'] {
  outline: 0;
}
```

Reply:
Lol, now I see it too 🤷‍♂️. Probably doing something wrong. Okay, I'll go ahead and remove, I'll do the same for Popover.

Reply:
I think we have some problems in the development environment, for some reason I can't see the rule in the development environment. I'll check this out later.